rmw_dds_common/test/test_graph_cache.cpp

```diff
@@ -1550,3 +1551,64 @@ TEST(test_graph_cache, bad_arguments)
     rcutils_reset_error();
   }
 }
+
+TEST(test_graph_cache, get_writers_info_by_topic_maybe_fail)
+{
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    GraphCache graph_cache;
+    add_participants(graph_cache, {"participant1"});
```

Comment:
Can this init section be left out of the fault_injection test macro?

Reply:
I think so. I created a test fixture class and split this test into three separate TEST_Fs